### PR TITLE
fix(sdk): remove contract verification check; add expectedimplementation; enhance logging

### PR DESCRIPTION
**Pull request by nbayindirli**
### Description

- removes contract verification check
- adds `expectedimplementation` method params for `verifyproxycontract`
- enhances logging and renames variables for clarity

### Drive-by changes

- none

### Related issues

- fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4123

### Backward compatibility

- yes

### Testing

- manual

### CODE REVIEW

File: typescript/sdk/src/deploy/verify/ContractVerifier.ts [hunk 1]
```diff
@@ -183,32 +191,39 @@ export class ContractVerifier {
...
-      return !!result[0]?.SourceCode;
-    } catch (error) {
-      verificationLogger.debug(
-        { error },
-        'Error checking if contract is already verified',
-      );
-      return false;
-    }
-  }
+  // private async isAlreadyVerified(
```
Comment (cmcewen) [inline typescript/sdk/src/deploy/verify/ContractVerifier.ts:201]: what is your goal in commenting this out? do you expect to use this again in the future?

File: typescript/sdk/src/deploy/verify/ContractVerifier.ts [hunk 2]
```diff
@@ -256,14 +267,22 @@ export class ContractVerifier {
     chain: ChainName,
     input: ContractVerificationInput,
     verificationLogger: Logger,
-  ): Promise<{ proxyGuid: any }> {
+  ): Promise<string> {
     try {
-      return this.submitForm(
+      const proxyGuid = await this.submitForm(
```
Comment (cmcewen) [inline typescript/sdk/src/deploy/verify/ContractVerifier.ts:272]: nit: i find it confusing that two different sounding functions (this.markProxy and this.submitForm) return a variable with the same name (proxyGuid)

File: typescript/sdk/src/deploy/verify/ContractVerifier.ts [hunk 3]
```diff
@@ -358,17 +377,17 @@ export class ContractVerifier {
...
-        chain,
-        input.address,
-      );
-      verificationLogger.debug(
-        `Contract already verified at ${addressUrl}#code`,
-      );
-      await sleep(200); // 5 calls/s (https://info.etherscan.com/api-return-errors/)
-      return;
-    }
+    // if (await this.isAlreadyVerified(chain, input, verificationLogger)) {
```
Comment (cmcewen) [inline typescript/sdk/src/deploy/verify/ContractVerifier.ts:380]: same comment about comment 😄

File: typescript/sdk/src/deploy/verify/ContractVerifier.ts (hunk 1 above)
Comment (nbayindirli) [reply typescript/sdk/src/deploy/verify/ContractVerifier.ts:201]: I expect us to reward the function in the future, but also commented code is dead code. I'll remove it and we can come back to this PR if we really need that code.

File: typescript/sdk/src/deploy/verify/ContractVerifier.ts (hunk 2 above)
Comment (nbayindirli) [reply typescript/sdk/src/deploy/verify/ContractVerifier.ts:245]: `submitForm()` only returns the response for whatever the request is (in this case, `ExplorerApiActions.VERIFY_PROXY`).

In the context of `markProxy()`, this is always `proxyGuid` if the response returns a 200

File: typescript/sdk/src/deploy/verify/ContractVerifier.ts (hunk 3 above)
Comment (nbayindirli) [reply typescript/sdk/src/deploy/verify/ContractVerifier.ts]: removed as well :)

**Pull request merged**